pcsx2/CMakeLists.txt

```diff
@@ -1222,6 +1222,8 @@ else()
 	target_link_libraries(PCSX2_FLAGS INTERFACE
 		OpenGL::GL
 		PCAP::PCAP
+		Wayland::Client
+		Wayland::Egl
```

Comment:
These shouldn't be here, it will link with wayland even if WAYLAND_API is OFF

Reply:
I suggest adding it to line 652
```
target_link_libraries(PCSX2_FLAGS INTERFACE
	Wayland::Client
	Wayland::Egl
)
```

Reply:
of this file

Reply:
Right on, without it it'll fail to link in OpenSuse so I'll take a look at those changes

Reply:
Why is this even here? We dlopen the Wayland egl lib specifically to avoid linking.

Reply:
Because in OpenSuse and similar environments it fails to link breaking SDL's vulkan.h where it includes wayland-client.h

`pcsx2/3rdparty/vulkan-headers/include/vulkan/vulkan.h:41:10: fatal error: wayland-client.h: No such file or directory
   41 | #include <wayland-client.h>
`

Reply:
That isn't a link error?

Reply:
Sorta. With the HINTS added to FIND_PATH it'll find all libs and includes and I can see it clear as day in the cache. Something else isnt linking to Wayland itself right so when it tries to include the header it'll give the error.

Reply:
Sorry, no, you're completely incorrect.

A symbol not being found/resolved is a link error.
A #include directive not finding a file is not a link error. The preprocessor runs several stages before the linker.